frontend/src/component/personalDashboard/useDashboardState.ts

```diff
@@ -35,34 +36,31 @@ export const useDashboardState = (
 
     useEffect(() => {
         const updates: Partial<State> = {};
-        const setDefaultFlag =
-            flags.length &&
-            (!state.activeFlag ||
-                !flags.some((flag) => flag.name === state.activeFlag?.name));
 
-        if (setDefaultFlag) {
-            updates.activeFlag = flags[0];
+        if (
+            props?.flags?.length &&
+            (!state.activeFlag ||
+                !props.flags.some(
+                    (flag) => flag.name === state.activeFlag?.name,
+                ))
```

Comment:
I took these out of variables because the linter didn't recognize `props.flags` as definitely being there if the variable was true. So to avoid using `!`, I put it in there.